eng/mgmt/automation/sdk_generate.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python3
+import os
+import re
+import sys
+import json
+import glob
+import logging
+import argparse
+
+pwd = os.getcwd()
+os.chdir(os.path.abspath(os.path.dirname(sys.argv[0])))
+from parameters import *
+from utils import set_or_increase_version
+from generate_data import sdk_automation as sdk_automation_data
+from generate_utils import (
+    compare_with_maven_package,
+    compile_package,
+    generate,
+    get_and_update_service_from_api_specs,
+    get_suffix_from_api_specs,
+)
+os.chdir(pwd)
+
+
+def update_parameters(suffix):
+    # update changeable parameters in parameters.py
+    global SUFFIX, NAMESPACE_SUFFIX, ARTIFACT_SUFFIX, NAMESPACE_FORMAT, ARTIFACT_FORMAT, OUTPUT_FOLDER_FORMAT
+
+    SUFFIX = suffix
+
+    NAMESPACE_SUFFIX = '.{0}'.format(SUFFIX) if SUFFIX else ''
+    ARTIFACT_SUFFIX = '-{0}'.format(SUFFIX) if SUFFIX else ''
+    NAMESPACE_FORMAT = 'com.azure.resourcemanager.{{0}}{0}'.format(
+        NAMESPACE_SUFFIX)
+    ARTIFACT_FORMAT = 'azure-resourcemanager-{{0}}{0}'.format(ARTIFACT_SUFFIX)
+    OUTPUT_FOLDER_FORMAT = 'sdk/{{0}}/{0}'.format(ARTIFACT_FORMAT)
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--spec-root',
+        default =
+        'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/',
+        help = 'Spec root folder',
+    )
+    parser.add_argument(
+        '-r',
+        '--readme',
+        help =
+        'Readme path, Sample: "storage" or "specification/storage/resource-manager/readme.md"',
+    )
+    parser.add_argument('-t', '--tag', help = 'Specific tag')
+    parser.add_argument('-v', '--version', help = 'Specific sdk version')
+    parser.add_argument(
+        '-s',
+        '--service',
+        help = 'Service Name if not the same as spec name',
+    )
```

Comment:
nit, not sure if you need these args in your automation script. Anyway, your script, yours to maintain.

Reply:
> nit, not sure if you need these args in your automation script. Anyway, your script, yours to maintain.

update with your changes is generate.py

Reply:
add it first for being compatible with existing functions and will discuss whether delete or change it